server/src/internalClusterTest/java/org/elasticsearch/action/bulk/WriteMemoryLimitsIT.java

```diff
@@ -210,16 +185,242 @@ public void testWriteBytesAreIncremented() throws Exception {
             if (replicationSendPointReached.getCount() > 0) {
                 replicationSendPointReached.countDown();
             }
-            while (newActionsSendPointReached.getCount() > 0) {
-                newActionsSendPointReached.countDown();
-            }
+            replicaRelease.close();
             if (latchBlockingReplicationSend.getCount() > 0) {
                 latchBlockingReplicationSend.countDown();
             }
-            if (latchBlockingReplication.getCount() > 0) {
-                latchBlockingReplication.countDown();
-            }
+            replicaRelease.close();
             primaryTransportService.clearAllRules();
         }
     }
+
+    public void testWriteCanBeRejectedAtCoordinatingLevel() throws Exception {
```

Comment:
As we discussed, we eventually want to have a test that also covers the rest layer.

Reply:
I added a comment.